CIP-1694/README.md

```diff
@@ -0,0 +1,624 @@
+Note also that a vote to abstain is different than abstaining from voting
+(the former is visible on chain, the latter is not).
+To avoid confusion, we will only use the word "abstain" from this point onward to mean a vote
+to abstain.
+
+The key hash will trigger the appropriate signature check on the transaction body according to the existing `UTxOW` ledger rule.
+
+Votes can be cast multiple times for each governance action by a single key hash. Correctly submitted votes override any older votes for the same key hash and role.
+As soon as a governance action is ratified, voting ends. Any future votes are not considered or recorded (whether they are yes, no or abstain).
+
+#### Governance State
+
+When a governance action is successfully submitted to the chain, its progress will be tracked by the ledger state.
+In particular, the following will be tracked:
+
+* the governance action ID;
+* the epoch that the action expires;
+* the deposit amount;
+* the rewards address that will receive the deposit when it is returned;
+* the total yes/no/abstain votes of the Constitutional Committee for this action;
+* the total yes/no/abstain votes of the DReps for this action;
+* the total yes/no/abstain votes of the SPOs  for this action.
+
+#### Stale Votes
+
+The votes from the DReps and the SPOs may become meaningless after crossing an epoch boundary since they may become unregistered. Therefore, all unregistered votes are cancelled before any new votes are considered.
+
+#### Changes to the Stake Snapshot
+
+Since the stake snapshot changes at each epoch boundary, a new current voting tally must be calculated for each governance action based on the votes that have been cast before any new votes are counted.  This avoids the same stake being used two or more times.  _This new tally may result in immediate ratification of a governance action._
+
+### Delegated Representatives (DReps)
+
+> **Warning**
+> The design of the DReps is still subject to change and should not be conflated with Project Catalyst DReps.
+
+Existing stake keys will be given new responsibilities. They will be able to delegate their stake to DReps. DRep Registration will mimic the existing stake delegation mechanisms (via certificates).
+
+The following new types of voting certificates will be added:
+
+* DRep registration certificate. Including:
+  * DRep ID: hash of verification key
+  * a URL for any metadata that is relevant to the DRep
+  * a hash of the contents of this URL
+
+* DRep retirement certificate. Including:
+  * DRep ID
+  * epoch number after which the DRep will retire
+
+* Vote delegation certificate. Including:
+  * stake credential
+  * DRep ID
+
+// TODO: Provide CBOR specification in the annexe for those new certificates.
+
+The authorization scheme (i.e. which signatures are required) mimics the existing stake delegation certificate authorization scheme.
+
+#### New Stake Distribution for DReps
```

Comment:
Has there been any discussion about partial stake delegation by governance action type? e.g. maybe someone wants to delegate stake for treasury actions and protocol parameter updates to one DRep and delegate to another DRep for actions relating the the CC and HF actions.

Reply:
yes, that's a very desirable feature, but it is unfortunately very disruptive to the current system. This would require adding more key (hashes) to the address structure, so we do not want to consider this for the first step towards Voltaire. See https://github.com/JaredCorduan/CIPs/blob/voltaire-v1/CIP-1694/README.md#piggybacking-on-stake-pool-stake-distribution

Reply:
Yeah, it certainly makes sense to weight implementation time vs feature completeness!